bootstrap/090_post_setup.sql

```diff
@@ -70,21 +70,84 @@ CREATE OR REPLACE TASK TASKS.WAREHOUSE_EVENTS_MAINTENANCE
     ALLOW_OVERLAPPING_EXECUTION = FALSE
     USER_TASK_MANAGED_INITIAL_WAREHOUSE_SIZE = "XSMALL"
     AS
-    CALL INTERNAL.refresh_warehouse_events(true);
+DECLARE
+    start_time timestamp_ltz default (select current_timestamp());
+    task_run_id text default (select INTERNAL.TASK_RUN_ID());
+    query_id text default (select query_id from table(information_schema.task_history(TASK_NAME => 'WAREHOUSE_EVENTS_MAINTENANCE')) WHERE GRAPH_RUN_GROUP_ID = :task_run_id  AND DATABASE_NAME = current_database() limit 1);
+    task_name text default 'WAREHOUSE_EVENTS_MAINTENANCE';
+    object_name text default 'WAREHOUSE_EVENTS_HISTORY';
+BEGIN
+    let input variant := (select output from INTERNAL.TASK_LOG where success AND task_name = :task_name AND object_name = :object_name order by task_start desc limit 1);
+    INSERT INTO INTERNAL.TASK_LOG(task_start, task_run_id, query_id, input, task_name, object_name) select :start_time, :task_run_id, :query_id, :input, :task_name, :object_name;
+
+    let output variant;
+    CALL INTERNAL.refresh_warehouse_events(true, :input) into :output;
+
+    let success boolean := (select :output['SQLERRM'] is null);
+    UPDATE INTERNAL.TASK_LOG SET success = :success, output = :output, task_finish = current_timestamp() WHERE task_start = :start_time AND task_run_id = :task_run_id;
+END;
 
 CREATE OR REPLACE TASK TASKS.SIMPLE_DATA_EVENTS_MAINTENANCE
     SCHEDULE = '60 minute'
     ALLOW_OVERLAPPING_EXECUTION = FALSE
     USER_TASK_MANAGED_INITIAL_WAREHOUSE_SIZE = "XSMALL"
     AS
-    CALL INTERNAL.refresh_all_simple_tables(true);
+DECLARE
+    task_name text default 'SIMPLE_DATA_EVENTS_MAINTENANCE';
+    task_run_id text default (select INTERNAL.TASK_RUN_ID());
+    query_id text default (select query_id from table(information_schema.task_history(TASK_NAME => 'SIMPLE_DATA_EVENTS_MAINTENANCE')) WHERE GRAPH_RUN_GROUP_ID = :task_run_id  AND DATABASE_NAME = current_database() limit 1);
```

Comment:
I'd love this to be a UDF; however, because this uses a context_function, it has to live in a task.

We could make a task DAG that ensures a task which creates a common UDF is run before any other task, but this felt overkill for the size of this PR already.

Reply:
I have a slight preference for this (and the other similar calls) to live in the `BEGIN`/`END` block. My general rule of thumb was to put only very simple statements into the `DECLAR` block. You do similar on eg line 142

Reply:
fixed (and swapped the string literal for the `:task_name` variable -- wasn't sure if that was going to work inside the UDTF argument list)